nixos/modules/services/monitoring/prometheus/alertmanager-irc-relay.nix

```
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.alertmanagerIrcRelay;
+
+  configFile = pkgs.writeText
+    "alertmanager-irc-relay.yml"
+    (builtins.toJSON cfg.configuration);
+in {
+  options.services.prometheus.alertmanagerIrcRelay = {
+    enable = mkEnableOption "Alertmanager IRC Relay";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.alertmanager-irc-relay;
+      defaultText = literalExpression "pkgs.alertmanager-irc-relay";
+      description = "Alertmanager IRC Relay package to use.";
+    };
+
+    configuration = mkOption {
```

```suggestion
    settings = mkOption {
```

Should follow RFC42 and likely use pkgs.formats.yaml.

---

Thanks for this feedback. I have applied it and just force pushed a newer version.